docs/devel/kubemark-guide.md

```diff
@@ -0,0 +1,108 @@
+
+To start a Kubemark cluster on GCE you need to create an external cluster (it can be GCE or GKE cluster) by yourself and run `test/kubemark/start-kubemark.sh` script. It will do all the setup for you using the configuration stored in `cluster/kubemark/config-default.sh` - you can tweak it however you want, but note that some features may not implemented yet, as implementation of Hollow components/mocks will probably be lagging behind ‘real’ one. For performance tests interesting variables are `NUM_MINIONS` and `MASTER_SIZE`. After start-kubemark script is finished you’ll have a ready Kubemark cluster, script stores a kubeconfig file for talking to the Kubemark cluster in `test/kubemark/kubeconfig.loc`.
+
+Currently single HollowNode uses 50 millicores of CPU and ~60MB or memory, which taking into account default cluster addons and fluentD, allows running ~17.5 HollowNodes per core.
+
+#### Behind the scene details:
+
+Start-kubemark script does quite a lot of things:
+- Creates a master machine called hollow-cluster-master and PD for it (uses gcloud, should be easy to do outside of GCE)
+- Creates a firewall rule which opens port 443 on the master machine (uses gcloud, should be easy to do outside of GCE)
+- Builds a Docker image for HollowNode from the current repository and pushes it to the Docker repository (GCR for us, using scripts from `cluster/gce/util.sh` - it may get tricky outside of GCE)
+- Generates certificates and kubeconfig files, writes a kubeconfig locally to `test/kubemark/kubeconfig.loc` and creates a Secret which stores kubeconfig for HollowKubelet/HollowProxy use (used gcloud to transfer files to Master, should be easy to do outside of GCE).
+- Creates a ReplicationController for HollowNodes and starts them up. (will work exactly the same everywhere as long as MASTER_IP will be populated correctly, but you’ll need to update docker image address if you’re not using GCR and default image name)
+
+### Running Density tests on Kubemark cluster
+
+To run standard density ‘30 test on your Kubemark cluster created in the previous step you execute `test/kubemark/run-scalability-test.sh` script. It will configure ginkgo to use Kubemark cluster instead of something else and start a density test. This script should not need any changes to work on other cloud providers.
+Monitoring test execution and debugging problems
+Run-scalability-test prints the same output on Kubemark as on ordinary e2e cluster, but if you need to dig deeper you need to learn how to debug HollowNodes and how Master machine (currently) differs from the ordinary one.
+
+If you need to debug master machine you can do similar things as you do on your ordinary master. The difference between Kubemark setup and ordinary setup is that in Kubemark etcd is run as a plain docker container, and all master components are run as normal processes. There’s no Kubelet overseeing them. Logs are stored in exactly the same place, i.e. `/var/logs/` directory.
```

Comment:
Not yet. The thing is that I need to decide how to handle 'real' Nodes for addons. It is tempting to just use `kube-up` script and add HollowNodes later on, but it can turn out to be quite delicate thing to do, as we'll need to have two kubernetes clusters and pods in one talking to the master of second one.

Hacking it to run as Pods alone should be doable, but I'll need to learn how cheat docker to prevent it from looking for images in the net.